python/pyiceberg/expressions/base.py

```diff
@@ -654,17 +664,37 @@ def _(obj: Or, visitor: BooleanExpressionVisitor[T]) -> T:
     return visitor.visit_or(left_result=left_result, right_result=right_result)
 
 
+def bind(schema: Schema, expression: BooleanExpression, case_sensitive: bool) -> BooleanExpression:
```

Comment:
Minor: we usually add a default for `case_sensitive`. I think it's a good idea to require it for internal methods, but this will probably be called from many different places so I'd default it here.